### PR TITLE
feat(agreement): add expiredIn virtual field to agreement model

### PR DESCRIPTION
**Pull request by nduchak**

### CODE REVIEW

File: src/services/storage/models/agreement.model.ts [hunk 1]
```diff
@@ -84,4 +84,15 @@ export default class Agreement extends Model {
   get hasSufficientFunds (): boolean {
     return this.availableFunds.minus(this.toBePayedOut).gte(this.periodPrice())
   }
+
+  /**
+   * Filed represent expiration time of agreement(in munutes)
+   */
+  @Column(DataType.VIRTUAL)
+  get expiredIn (): BigNumber {
```
Comment (jurajpiar) [inline src/services/storage/models/agreement.model.ts:92]: expiresIn would be better

File: src/services/storage/models/agreement.model.ts [hunk 2]
```diff
@@ -84,4 +84,15 @@ export default class Agreement extends Model {
   get hasSufficientFunds (): boolean {
     return this.availableFunds.minus(this.toBePayedOut).gte(this.periodPrice())
   }
+
+  /**
+   * Filed represent expiration time of agreement(in munutes)
```
Comment (jurajpiar) [inline src/services/storage/models/agreement.model.ts:89]:
```suggestion
   * Field represents the time (in minutes) until the agreement expires
```

File: src/services/storage/models/agreement.model.ts [hunk 3]
```diff
@@ -84,4 +84,15 @@ export default class Agreement extends Model {
...
+   */
+  @Column(DataType.VIRTUAL)
+  get expiredIn (): BigNumber {
+    if (!this.hasSufficientFunds) return new BigNumber(0)
+    const availableFundsAfterPayout = this.availableFunds.minus(this.toBePayedOut)
+
+    return bnFloor(availableFundsAfterPayout.div(this.periodPrice())).times(this.billingPeriod.div(60)) // in minutes
```
Comment (jurajpiar) [inline src/services/storage/models/agreement.model.ts:96]: Would be more useful in seconds for the UI.

File: src/services/storage/models/agreement.model.ts [hunk 4]
```diff
@@ -84,4 +84,15 @@ export default class Agreement extends Model {
...
+
+  /**
+   * Field represents the time (in seconds) until the agreement expires
+   */
+  @Column(DataType.VIRTUAL)
+  get expiresIn (): BigNumber {
+    if (!this.hasSufficientFunds) return new BigNumber(0)
+    const availableFundsAfterPayout = this.availableFunds.minus(this.toBePayedOut)
+
+    return bnFloor(availableFundsAfterPayout.div(this.periodPrice())).times(this.billingPeriod) // in minutes
```
Comment (jurajpiar) [inline src/services/storage/models/agreement.model.ts:96]: maybe remove this comment

File: src/services/storage/models/agreement.model.ts (hunk 2 above)
Comment (nduchak) [reply src/services/storage/models/agreement.model.ts]: Done

File: src/services/storage/models/agreement.model.ts (hunk 4 above)
Comment (nduchak) [reply src/services/storage/models/agreement.model.ts:96]: Done

File: src/services/storage/models/agreement.model.ts (hunk 3 above)
Comment (nduchak) [reply src/services/storage/models/agreement.model.ts]: Done

**Review by jurajpiar — APPROVED** [review]
LGTM

**Review by AuHau — APPROVED** [review]
LGTM, I thought originally we already had something like this, but I guess I forgot 😅

**Pull request merged**